startup.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import sgtk
+import xml.etree.ElementTree as XET
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class MayaLauncher(SoftwareLauncher):
+    """
+    Handles launching Maya in an engine-agnostic manner.
+    """
+
+    def synergy_paths(self):
+        """
+        Override base class definition to restrict list of Synergy
+        Config files to be Maya-specific
+        """
+        if self._synergy_paths is None:
+            # Let the parent class deal with finding the files
+            all_paths = SoftwareLauncher.synergy_paths(self)
+            # Now pick out the specific ones for Maya.
+            self._synergy_paths = [
+                p for p in all_paths if os.path.basename(p).startswith("Maya")
+            ]
+        return self._synergy_paths
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param versions: List of strings representing versions
+                         to search for. If set to None, search
+                         for all versions. A version string is
+                         DCC-specific but could be something
+                         like "2017", "6.3v7" or "1.2.3.52"
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        sw_versions = []
+        for synergy_cfg in self.synergy_paths():
+            swv = self._software_version_from_synergy(synergy_cfg)
+            if not swv:
+                self.logger.debug(
+                    "Could not create SoftwareVersion from Synergy config '%s'" %
+                    (synergy_cfg)
+                )
+                continue
+            if versions and isinstance(versions, list) and str(swv.version) not in versions:
+                self.logger.debug(
+                    "Resolved version for Synergy config '%s' [%s] not found "
+                    "in specified list of versions : %s" %
+                    (synergy_cfg, swv.version, versions)
+                )
+                continue
+            sw_versions.append(swv)
+
+        [self._customize_software_version(swv) for swv in sw_versions]
```

Comment:
this approach seems a little odd - i think it might be simpler if all the data is determined up front and fed into the software version (which wouldn't have any setters). Maybe it's easier if the synergy stuff is pulled out of core and put into the maya engine too?